python-frontend/src/main/java/org/sonar/python/index/ProjectDescriptor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.index;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.sonar.plugins.python.api.PythonFile;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.symbols.Usage;
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.python.semantic.SymbolTableBuilder;
+import org.sonar.python.semantic.SymbolUtils;
+
+public class ProjectDescriptor {
+  private final Map<String, ModuleDescriptor> modules = new HashMap<>();
+
+  public void addModule(FileInput fileInput, String packageName, PythonFile pythonFile) {
+    SymbolTableBuilder symbolTableBuilder = new SymbolTableBuilder(packageName, pythonFile);
+    String fullyQualifiedModuleName = SymbolUtils.fullyQualifiedModuleName(packageName, pythonFile.fileName());
+    fileInput.accept(symbolTableBuilder);
+
+    List<Descriptor> descriptors = fileInput.globalVariables().stream()
+      // TODO: We don't put builtin or imported names in global symbol table to avoid duplicate FQNs in project level symbol table (to fix with SONARPY-647)
```

Comment:
I think that eventually it's a very good idea to store "aliases" information and indeed it could be done in this class.
I'm not so sure about naming this "NamespaceDescriptor" because if I understood correctly "namespace" in Python is pretty vague term and also a `class` creates a namespace.
So I would tend to call this class either `ModuleDescriptor` or `FileDescriptor` because I want it to be associated with a Python file. Does it make sense?